tests/core/base/test_prompt.py

```diff
@@ -55,6 +56,35 @@ def genre(self) -> str:
     assert str(prompt) == "Recommend a fantasy book."
 
 
+def test_base_prompt_with_prompt_template() -> None:
+    """Tests the `BasePrompt` class with prompt_template."""
+
+    class BookRecommendationPrompt(BasePrompt):
+        prompt_template: ClassVar[str] = "Recommend a {genre} book."
```

Comment:
can we also test setting it without having to write `ClassVar[str]`?

For example:

```python
class BookRecommendationPrompt(BasePrompt):
    prompt_template = "recommend a {genre} book"
    genre: str
```

I want to make sure this works properly so that users can more easily define prompts with fewer imports